dotnet/src/Planners/Planners.Core/Plan.cs

```diff
@@ -371,9 +412,9 @@ private async Task<FunctionResult> InternalInvokeNextStepAsync(Kernel kernel, Co
             var functionVariables = this.GetNextStepVariables(variables, step);
 
             // Execute the step
-            var result = await kernel.InvokeAsync(step, functionVariables, cancellationToken).ConfigureAwait(false);
+            var result = await step.InvokeAsync(kernel, functionVariables, null, cancellationToken).ConfigureAwait(false);
 
-            var resultValue = result.Variables.Input.Trim();
+            var resultValue = result.GetValue<string>()?.Trim();
```

Comment:
The line will throw the `InvalidCastException` for values of non-string type. For example, if the function returned integer - 23 the GetValue method will throw - System.InvalidCastException: 'Cannot cast System.Int32 to System.String'

Reply:
Will change so that the code does something that is functionally equivalent.